src/main.rs

```diff
@@ -270,9 +305,21 @@ fn run(mut cli: Cli) -> Result<(), Box<dyn std::error::Error>> {
             println!("{}", serde_json::to_string_pretty(&meta)?);
         }
     } else {
-        let commands = CommandSource::new(config.bindings.clone())?;
+        let speaker_notes_event_receiver = if let Some(SpeakerNotesMode::Receiver) = cli.speaker_notes_mode {
+            let receiver = create_speaker_notes_service_builder(&path)?.open()?.subscriber_builder().create()?;
```

Comment:
This should fail with a nice looking error. I currently only see 

```
PublishSubscribeOpenError::DoesNotExist
````

When I run the receiver before the publisher. Maybe use a `Cli::command().error(...)` like above in this file with a meaningful error message like "no presenterm in publisher mode running" or something like that.

Same for the publisher failing when the channel is already open.

Reply:
Agreed.
There are several different error enums - each with multiple variants - involved in this chain of method calls, which makes this tricky to isolate these particular variants.

I'll see what I can do.

Reply:
I have added a custom `IpcServiceError` enum and `From` implementations to convert the specific `iceoryx2` error enum variants for these two specific scenarios, in order to provide better error messages. This could be extended to cover more error cases if we wished. Let me know what you think.